src/index.js

```diff
@@ -47,15 +48,27 @@ function init (companyId, apiKey, token) {
 class AnalyticsCoreAPI {
   /** Initialize sdk.
   *
-  * @param companyId {string} company ID to be used with Adobe Analytics.
+  * @param companyId {string} company ID to be used with Adobe Analytics (optional - if passed as null, then getDiscoveryCredentials is called to retrieve the company ID).
   * @param apiKey {string} Your api key
   * @param token {string} Valid auth token
+  * @param options {object} Extensible to include additional parameters, as necessary.  For example, can be used to include valid IMS Org ID (i.e., 'F9452D1224FEA71029B5C8A97@AdobeOrg' ) which is only passed when a companyId is unavaiable.
   * @returns {AnalyticsCoreAPI}
   */
-  async init (companyId, apiKey, token) {
+  async init (companyId, apiKey, token, options) {
     const initErrors = []
-    if (!companyId) {
-      initErrors.push('companyId')
+    if (!companyId || companyId === null) {
+      const discoveryResponse = await this.getDiscoveryCredentials(apiKey, token)
+      for (const i in discoveryResponse.imsOrgs) {
+        if (options !== null || options !== undefined) {
+          if (options.imsOrg !== null || options.imsOrg !== undefined) {
+            if (discoveryResponse.imsOrgs[i].imsOrgId === options.imsOrg) {
+              companyId = discoveryResponse.imsOrgs[i].companies[0].globalCompanyId
+              var companyName = discoveryResponse.imsOrgs[i].companies[0].companyName
```

Comment:
use `let` not `var`, it is a better practice and more clear where the scope of the variable is. declare it in the `init` function scope, then assign it here.